pupepat/test/pupil-synth/src/source.py

```diff
@@ -0,0 +1,157 @@
+import abc
+import numpy as np
+
+
+class Source(abc.ABC):
+    def __init__(self, x=None, y=None):
+        """
+        A Source need not know it's position on a plate until it's been added.
+        :param x:
+        :param y:
+        """
+        self.x, self.y = x, y
+        self.set_position(x, y)
+        self.half_width = 0  # decal.shape[0] = 2*half_width + 1
+        super().__init__()
+
+    def set_position(self, x, y):
+        self.x = x
+        self.y = y
+
+    @property
+    def position(self):
+        return self.x, self.y
+
+    @property
+    @abc.abstractmethod
+    def decal(self):
+        pass
+
+
+class PointSource(Source):
```

Comment:
Technically this is not a point source. A point source would have roughly a Gaussian profile (or whatever the Point Spread Function is). This is more of a circular source.